docs/ops/sequence/CTCLoss_4.md

```diff
@@ -0,0 +1,125 @@
+\f[
+p_{t,i,j} = \frac{\exp(logits[t,i,j])}{\sum^{K}_{k=0}{\exp(logits[t,i,k])}}
+\f]
+
+2. For a given `i`-th target from `labels[i,:]` find all aligned paths.
+A path `S = (c1,c2,...,cT)` is aligned with a target `G=(g1,g2,...,gT)` if both chains are equal after decoding.
+The decoding extracts substring of length `label_length[i]` from a target `G`, merges repeated characters in `G` in case *preprocess_collapse_repeated* equal to True and
+finds unique elements in the order of character occurence in case *unique* equal to True.
+The decoding merges repeated characters in `S` in case *ctc_merge_repeated* equal to True and removes blank characters represented by `blank_index`.
+By default, `blank_index` is equal to `C-1`, where `C` is a number of classes including the blank.
+For example, in case default *ctc_merge_repeated*, *preprocess_collapse_repeated*, *unique*, and `blank_index` a target sequence `G=(0,3,2,2,2,2,2,4,3)` of a length `label_length[i]=4` is processed 
+to `(0,3,2,2)` and a path `S=(0,0,4,3,2,2,4,2,4)` of a length `logit_length[i]=9` is also processed to `(0,3,2,2)`, where `C=5`.
+There exist other paths that are also aligned with `G`, for instance, `0,4,3,3,2,4,2,2,2`. Paths checked for alignment with a target `label[:,i]` must be of length `logit_length[i] = L_i`.
+Compute probabilities of these aligned paths (alignments) as follows:
+\f[
+p(S) = \prod_{t=1}^{L_i} p_{t,i,ct}
+\f]
+
+3. Finally, compute negative sum of log-probabilities of all found alignments:
+\f[
+CTCLoss = \minus \sum_{S} \ln p(S)
+\f]
+
+**Note**:  This calculation scheme does not provide steps for optimal implementation and primarily serves for better explanation.
+
+**Attributes**
+
+* *preprocess_collapse_repeated*
+
+  * **Description**: *preprocess_collapse_repeated* is a flag for a preprocessing step before loss calculation, wherein repeated labels in `labels[i,:]` passed to the loss are merged into single labels.
+  * **Range of values**: True or False
+  * **Type**: `boolean`
+  * **Default value**: False
+  * **Required**: *no*
+
+* *ctc_merge_repeated*
+
+  * **Description**: *ctc_merge_repeated* is a flag for merging repeated characters in a potential alignment during the CTC loss calculation.
+  * **Range of values**: True or False
+  * **Type**: `boolean`
+  * **Default value**: True
+  * **Required**: *no*
+
+* *unique*
+
+  * **Description**: *unique* is a flag to find unique elements for a target `labels[i,:]` before matching with potential alignments.
+  * **Range of values**: True or False
+  * **Type**: `boolean`
+  * **Default value**: False
+  * **Required**: *no*
+
+**Inputs**
+
+* **1**: `logits` - Input tensor with a batch of sequences of logits. Type of elements is *T_F*. Shape of the tensor is `[T, N, C]`, where `T` is the maximum sequence length, `N` is the batch size and `C` is the number of classes including the blank. Required.
```

Comment:
It is likely a problem that batch dimension is not the first in a shape. I remember a discussion on SetBatch being broken by CTCGreedyDecoder op because batch is not the first there and ngraph has no explicit notion of data layout. Also seems inconsistent with Labes input which is [N,T]

Reply:
Agree. Actually, the layout from `CTCGreedyDecoder` confused me and I thought such layout has benefits in terms of performance.
If no objectives from plugin owners, I am leaning to change it.

Reply:
Ask Sergey Lyalin about optimal layout for `logits`.

Reply:
Changed layout for `logits`.